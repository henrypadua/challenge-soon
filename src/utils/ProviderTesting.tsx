import { NativeBaseProvider, NativeBaseProviderProps } from 'native-base';

import { theme } from '@styles/theme';

export const ProviderTesting = (props: NativeBaseProviderProps) => {
	const inset = {
		frame: { x: 0, y: 0, width: 0, height: 0 },
		insets: { top: 0, left: 0, right: 0, bottom: 0 },
	};

	return (
		<NativeBaseProvider initialWindowMetrics={inset} theme={theme}>
			{props.children}
		</NativeBaseProvider>
	);
};
