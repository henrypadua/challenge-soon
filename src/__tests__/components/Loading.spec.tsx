import { render } from '@testing-library/react-native';

import { ProviderTesting } from '@utils/ProviderTesting';

import { Loading } from '@components/Loading';

jest.useFakeTimers();

test('renders correctly', () => {
	const screen = render(
		<ProviderTesting>
			<Loading />
		</ProviderTesting>,
	).toJSON();

	expect(screen).toMatchSnapshot();
});
