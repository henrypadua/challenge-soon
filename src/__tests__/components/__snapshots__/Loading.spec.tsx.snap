// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  dataSet={{}}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#3f3f46",
      "display": "flex",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    accessibilityLabel="loading"
    accessible={true}
    color="#be185d"
    size="large"
    style={
      [
        [
          {},
          {
            "dataSet": {},
          },
          {},
        ],
        undefined,
      ]
    }
  />
</View>
`;
